### PR TITLE
[Test-Proxy] Allow debug logging with docker host from CI. Bump targeted test-proxy version. Update versioning methodology.

### PR DESCRIPTION
**Pull request by scbedd**
Related to #2769

### CODE REVIEW

File: eng/common/testproxy/docker-start-proxy.ps1 [hunk 1]
```diff
@@ -31,11 +36,15 @@ catch {
     Write-Error "Please check your docker invocation and try running the script again."
 }
 
-$SELECTED_IMAGE_TAG = "1369319"
+$SELECTED_IMAGE_TAG = "1.0.0-dev20220224.1"
```
Comment (benbp) [inline eng/common/testproxy/docker-start-proxy.ps1:39]: This also works if you want to be more concise (not necessarily better):

```suggestion
$SELECTED_IMAGE_TAG = $VersionOverride ?? "1.0.0-dev20220224.1"
```

File: eng/common/testproxy/docker-start-proxy.ps1 [hunk 2]
```diff
@@ -51,15 +60,21 @@ function Get-Proxy-Container(){
...
 if ($IsWindows -and $env:TF_BUILD){
     $SelectedImage = $WINDOWS_IMAGE_SOURCE
     $Initial = "C:"
-    $LinuxContainerArgs = ""
+    $AdditionalContainerArgs = ""
+}
+
+# there isn't really a great way to get environment variables automagically from the CI environment to the docker image
+# handle loglevel here so that setting such a setting in CI will also bump the docker logging
+if ($env:Logging__LogLevel__Default){
```
Comment (benbp) [inline eng/common/testproxy/docker-start-proxy.ps1:76]: Nit:

```suggestion
if ($env:Logging__LogLevel__Default) {
```

**Review by benbp — APPROVED** [review]

File: eng/common/testproxy/docker-start-proxy.ps1 (hunk 1 above)
Comment (scbedd) [reply eng/common/testproxy/docker-start-proxy.ps1:39]: The reason I didn't do it this way is that folks are currently pulling this value from the script. EG: Python and JS both spin up the test-proxy as part of their stuff, they parse this line of code to pull the version. 

Maybe I move it to a single file that people can just read to get the version? Then it's a breaking change, but won't ever change in the future.

Comment (benbp) [reply eng/common/testproxy/docker-start-proxy.ps1:39]: Ahh I see. Yeah with little context it seems to me like keeping the version config separate from code, especially if parsed, is a good approach if you want to change it that way.

Comment (scbedd) [reply eng/common/testproxy/docker-start-proxy.ps1:39]: I  am just going to bite the bullet and deal with a breaking change now.

Comment (scbedd) [reply eng/common/testproxy/docker-start-proxy.ps1]: @benbp I have converted to version off the file. Will patch the `js` and `python` pipelines in the sync PRs when I approve this to the next stage.

Comment (scbedd) [reply eng/common/testproxy/docker-start-proxy.ps1]: Gave me the kick in the pants I needed.

File: eng/common/testproxy/docker-start-proxy.ps1 [hunk 3]
```diff
@@ -31,11 +36,16 @@ catch {
     Write-Error "Please check your docker invocation and try running the script again."
 }
 
-$SELECTED_IMAGE_TAG = "1369319"
+$SELECTED_IMAGE_TAG = $(Get-Content "$PSScriptRoot/target_version.txt" -Raw).Trim()
```
Comment (benbp) [inline eng/common/testproxy/docker-start-proxy.ps1:39]: `git add target_version.txt` ?

File: eng/common/testproxy/test-proxy-tool.yml [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
   runProxy: true
+  targetVersion: ''
 
 steps:
   - pwsh: |
         $(Build.SourcesDirectory)/eng/common/scripts/trust-proxy-certificate.ps1
     displayName: 'Language Specific Certificate Trust'
 
   - pwsh: |
+      $version = $(Get-Content C:\repo\sdk-for-python\eng\common\testproxy\target_version.txt -Raw).Trim()
```
Comment (benbp) [inline eng/common/testproxy/test-proxy-tool.yml:12]: Is the parameter necessary since you're doing the lookup in the script?

Comment (scbedd) [reply eng/common/testproxy/test-proxy-tool.yml:12]: YEs. Allowing override.

**Review by benbp — APPROVED** [review]

**Pull request merged**